fsspec/implementations/dbfs.py

```diff
@@ -1,66 +1,115 @@
+
         self.session = requests.Session()
         self.session.headers.update(self.header)
+
         super().__init__(**kwargs)
 
     @property
     def header(self):
         return {"Authorization": f"Bearer {self.token}"}
 
     @property
-    def url(self)
-        return f"https://{self.instance}/api/2.0/clusters/list"
+    def url(self):
+        return f"https://{self.instance}/api"
 
     def ls(self, path, detail=True):
         out = self._ls_from_cache(path)
         if not out:
-            r = self.session.get(self.url + "/2.0/dbfs/list", json={"path": path})
-            r.raise_for_status()
-            out = [{"name": o["path"],
-                    "type": ["directory", "file"][o["is_dir"]],
-                    "size": o["file_size"]}
-                   for o in r.json()["files"]]
+            try:
+                r = self.get_from_api("list", json={"path": path})
+            except requests.exceptions.HTTPError as e:
+                status_code = e.response.status_code
+                if status_code == 404:
+                    raise FileNotFoundError(path)
+                raise e
+            files = r["files"]
+            out = [
+                {
+                    "name": o["path"],
+                    "type": "directory" if o["is_dir"] else "file",
+                    "size": o["file_size"],
+                }
+                for o in files
+            ]
             self.dircache[path] = out
+
         if detail:
             return out
         return [o["name"] for o in out]
 
     def mkdirs(self, path, exist_ok=True):
         if not exist_ok:
             raise NotImplementedError
-        r = self.session.post(self.url + "/2.0/dbfs/mkdirs", json={"path": path})
-        r.raise_for_status()
+        self.post_to_api("mkdirs", json={"path": path})
 
-    def makedir(self, path, create_parents=True, **kwargs):
+    def makedirs(self, path, create_parents=True, **kwargs):
         if not create_parents:
             raise NotImplementedError
-        self.mkdirs(path)
+        self.mkdirs(path, **kwargs)
 
     def rm(self, path, recursive=False):
-        r = self.session.post(self.url + "/2.0/dbfs/delete",
-                              json={"path": path, "recursive": recursive})
-        r.raise_for_status()
+        self.post_to_api("delete", json={"path": path, "recursive": recursive})
 
-    def mv(self, path1, path2, recursive=False, maxdepth=None, **kwargs):
+    def mv(self, path1, path2, recursive=False, maxdepth=None):
         if recursive:
             raise NotImplementedError
-        r = self.session.post(self.url + "/2.0/dbfs/move",
-                              json={"source_path": path1, "destination_path": path2})
+        if maxdepth:
+            raise NotImplementedError
+
+        self.post_to_api(
+            "move",
+            json={"source_path": path1, "destination_path": path2},
+        )
+
+    def create_handle(self, path, overwrite=True):
```

Comment:
These nicely reflect the real API, but are not meant to be called directly by users, so might prefix with underscore